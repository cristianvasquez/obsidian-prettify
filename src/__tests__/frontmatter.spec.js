const prettifier = require('../prettifier')
const moment = require('moment')
const fixed_date = moment('2010-06-09T15:20:00-07:00')


test("Does not create a header", () => {
    const content = `No Header`
    return prettifier(content, {createHeaderIfNotPresent: false}, {}, {today: fixed_date}).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Creates a header", () => {
    const content = `No Header`
    return prettifier(content,
        {
            createHeaderIfNotPresent: true
        }, {today: fixed_date}
    ).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Creates date first time", () => {
    const content = `No Header`
    return prettifier(content,
        {
            createHeaderIfNotPresent: true,
            updateHeader: false,
        }, {today: fixed_date}).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Does not modify date", () => {
    const content = `---
date updated: '1999-06-10T00:20:00+02:00'

---`
    return prettifier(content, {
        updateHeader: false,
    }, {today: fixed_date}).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Maintains properties in header", () => {
    const content = `---
title: 'Example'
list:
  - oneX
  - 0
  - false
---
# What the Semantic Web ignores
    
The Web is not only about facts,
It's about humans communicating all sorts of things.
In their own language, with their own model of the world.

perhaps 1% ? of the humans use it for facts?
            `
    return prettifier(content, {}
        , {today: fixed_date}).then(data => {
        expect(data).toMatchSnapshot();
    });
});


test("Respects hash", () => {
    const content = `---
tag: '#Interpretability'
---
            `
    return prettifier(content, {},
        {today: fixed_date}).then(data => {
        expect(data).toMatchSnapshot();
    });
});

test("Respects emojis", () => {
    const content = `---
tag: '#🦐'
---
🦐
            `
    return prettifier(content, {}, {today: fixed_date}).then(data => {
        expect(data).toMatchSnapshot();
    });
});


test("add quote", () => {
    const content = `---
tag: #hello
---
            `
    return prettifier(content, {}, {
        frontMatterData: {}
    }).then(data => {
        expect(data).toMatchSnapshot();
    });
});

it("Add quotes", () => {
    const content = `---
Name: 
 - #Hello
 - World
 - ' #Hi'
 - #with/tab
 - #with/tab/teb/tib
 - #howdy_bla
 - #howdy-bla
Other tag: #Nothing
tags: #Nothing
---

Outside frontmatter

#tag

`;
    return prettifier(content, {}, {today: fixed_date}).then(data => {
        expect(data).toMatchSnapshot();
    });
});


it("Adds tags", () => {
    const content = `---
Name: 
 - #Hi
---


`;
    return prettifier(content,
        {}, {today: fixed_date, tags: ['#hello', '#world']}).then(data => {
        expect(data).toMatchSnapshot();
    });
});

it("Preserve previous tags", () => {
    const content = `---
tags: 
 - #preserve_me
---


`;
    return prettifier(content,
        {}, {today: fixed_date, tags: ['#hello', '#world']}).then(data => {
        expect(data).toMatchSnapshot();
    });
});



it("Preserve previous tags", () => {
    const content = `---
date updated: '2020-12-09T00:14:06+01:00'

tags: '#literatura'

---
`
    return prettifier(content,
        {}, {today: fixed_date, tags: []}).then(data => {
        expect(data).toMatchSnapshot();
    });
});

