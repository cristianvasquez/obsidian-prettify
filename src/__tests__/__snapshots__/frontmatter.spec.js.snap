// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creates a header 1`] = `
VFile {
  "contents": "---
date updated: '2010-06-10T00:20:00+02:00'

---

No Header
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Creates date first time 1`] = `
VFile {
  "contents": "---
date updated: '2010-06-10T00:20:00+02:00'

---

No Header
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Does not create a header 1`] = `
VFile {
  "contents": "No Header
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Does not modify date 1`] = `
VFile {
  "contents": "---
date updated: '1999-06-10T00:20:00+02:00'

---
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Maintains properties in header 1`] = `
VFile {
  "contents": "---
title: Example
list:
  - oneX
  - 0
  - false
date updated: '2010-06-10T00:20:00+02:00'

---

# What the Semantic Web ignores

The Web is not only about facts,
It's about humans communicating all sorts of things.
In their own language, with their own model of the world.

perhaps 1% ? of the humans use it for facts?
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Respects emojis 1`] = `
VFile {
  "contents": "---
tag: '#ü¶ê'
date updated: '2010-06-10T00:20:00+02:00'

---

ü¶ê
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;

exports[`Respects hash 1`] = `
VFile {
  "contents": "---
tag: '#Interpretability'
date updated: '2010-06-10T00:20:00+02:00'

---
",
  "cwd": "/home/cvasquez/obsidian/workspace/.obsidian/plugins/markdown-prettifier",
  "data": Object {},
  "history": Array [],
  "messages": Array [],
}
`;
